Content/Items/Dungeon/Weapons.ThousandthDegree.cs

```diff
@@ -0,0 +1,698 @@
+        {
+            target.AddBuff(BuffID.OnFire, 240);
+            target.AddBuff(BuffID.OnFire3, 240);
+
+            Helper.PlayPitched("Impacts/FireBladeStab", 0.2f, Main.rand.NextFloat(-0.05f, 0.05f), Projectile.Center);
+        }
+
+        public override void Kill(int timeLeft)
+        {
+            for (int i = 0; i < 35; ++i)
+            {
+                float angle2 = 6.2831855f * (float)i / (float)35;
+                Dust dust3 = Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(255, 50, 15), 0.6f);
+                dust3.velocity = Utils.ToRotationVector2(angle2) * 2.75f;
+                dust3.noGravity = true;
+            }
+
+            for (int i = 0; i < 35; ++i)
+            {
+                float angle2 = 6.2831855f * (float)i / (float)35;
+                Dust dust3 = Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(255, 160, 50), 0.45f);
+                dust3.velocity = Utils.ToRotationVector2(angle2) * 2f;
+                dust3.noGravity = true;
+            }
+        }
+        public override bool OnTileCollide(Vector2 oldVelocity)
+        {
+            if (!returning)
+                collided = true;
+            return false;
+        }
+
+        protected virtual Vector2 Collide()
+        {
+            return Collision.noSlopeCollision(Projectile.position, Projectile.velocity, Projectile.width, Projectile.height, true, true);
+        }
+
+        private void VanillaBoomerangAI() //this is bad vanilla code but I tried to make it as readable as I could
+        {
+            Vector2 wheelPosition = ((ThousandthDegreeProjectile)parentProj.ModProjectile).wheelPos;
+            Vector2 pos = new Vector2(Projectile.position.X + (float)Projectile.width * 0.5f, Projectile.position.Y + (float)Projectile.height * 0.5f);
+            float num = wheelPosition.X - pos.X;
+            float num2 = wheelPosition.Y - pos.Y;
+            float num3 = (float)Math.Sqrt(num * num + num2 * num2);
+            float speed = num3 > 450f ? 17f : 12f;
+            float adjust = 1f;
+            if (num3 > 3000f && lerpTimer == 0)
+                lerpTimer = 30;
+            num3 = speed / num3;
+            num *= num3;
+            num2 *= num3;
+            if (Projectile.velocity.X < num)
+            {
+                Projectile.velocity.X += adjust;
+                if (Projectile.velocity.X < 0f && num > 0f)
+                    Projectile.velocity.X += adjust;
+            }
+            else if (Projectile.velocity.X > num)
+            {
+                Projectile.velocity.X -= adjust;
+                if (Projectile.velocity.X > 0f && num < 0f)
+                    Projectile.velocity.X -= adjust;
+            }
+            if (Projectile.velocity.Y < num2)
+            {
+                Projectile.velocity.Y += adjust;
+                if (Projectile.velocity.Y < 0f && num2 > 0f)
```

Comment:
Never leave variables named ``numX`` in adapted code -- rename these variables to something readable